### PR TITLE
DGGTweaker Features - Link Hitbox Scaling, RustleSearch button and Lucide Icons

### PR DESCRIPTION
**Pull request by AlexDavies8**
After Cake mentioned it in the discord I figured I'd port the chat related stuff from [my extension](https://github.com/AlexDavies8/dgg-tweaker) to the main chat codebase.
Here's a list of the changes:
- Added a new chat setting to increase the clickable area of a link (bigger 'hitbox')
- Added button to the user info right click menu that directly links to rustlesearch.dev
- Basic support for numeric fields in chat settings (in the future could put maxlines in the UI instead of needing a command)
- Icons have been swapped out for Lucide icons to match with the main site redesign
- Button tooltips have been changed from using the title attribute to using tippy, and the tippy style is prettier

New Look

### CODE REVIEW

File: assets/chat/img/icon-open-in-new.svg
Comment (11k) [inline assets/chat/img/icon-open-in-new.svg:1]: I don't think this icon is used anywhere. If so, it's safe to delete.

File: assets/chat/img/icon-favorite-fill.svg
Comment (11k) [inline assets/chat/img/icon-favorite-fill.svg:1]: It seems new favorite emote icons are clipped. Would you mind fixing this?

**Review by 11k — CHANGES_REQUESTED** [review]
For your link scaling setting, (1) the way it's presented in the settings menu is confusing to the average user, (2) there's too much flexibility in allowing a custom multiplier considering only a few values are actually sane, and (3) the scaling seems to apply upward more than it does downward.

We can work to address these issues, but I don't think it's worth it because difficulty with clicking links isn't a big problem. The setting works perfectly as a customization for interested users to apply via an extension like yours.

Other than that, would you mind updating the embed link icon to Lucide as well? It's the TV that's displayed alongside an embeddable URL and is defined inline in `UrlFormatter.js`.

File: assets/chat/js/menus/ChatUserMenu.js [hunk 1]
```diff
@@ -244,7 +244,7 @@ export default class ChatUserMenu extends ChatMenu {
     const features =
       user.features.length === 0 ? 'nofeature' : user.features.join(' ');
     const usr = $(
-      `<div class="user-entry" data-username="${user.username}" data-user-id="${user.id}"><span class="user ${features}">${label}</span><div class="user-actions"><i class="mention-nick"></i><i class="whisper-nick"></i></div></div>`,
+      `<div class="user-entry" data-username="${user.username}" data-user-id="${user.id}"><span class="user ${features}">${label}</span><div class="user-actions"><i class="mention-nick" title="Mention"></i><i class="whisper-nick" title="Whisper"></i></div></div>`,
```
[the remark this answers is not in the document]
Comment (AlexDavies8) [reply assets/chat/js/menus/ChatUserMenu.js]: I tried this but they didn't get registered properly. It should now work with tippy, as I'm manually registering them with tippy.

File: assets/chat/img/icon-open-in-new.svg
Comment (AlexDavies8) [reply assets/chat/img/icon-open-in-new.svg:1]: Done 👍

File: assets/chat/img/icon-favorite-fill.svg
Comment (AlexDavies8) [reply assets/chat/img/icon-favorite-fill.svg:1]: Done 👍

**Comment by AlexDavies8** [comment]
I've reverted the link hitbox changes, changed the embed icon and addressed other mentioned changes.

**Review by 11k — APPROVED** [review]

**Pull request merged**